### DISCUSSION TITLE
Difference between meanOrientation and ODF in Pole figures

### DISCUSSION THREAD

**Discussion by Jesusmunozmontecinos** [mtex-toolbox/mtex#864] [Ask Anything]
Dear all,

Inspired in the recent MTEX workshop, I'm wondering what is the difference when plotting Pole figures based on an ODF or using the meanOrientation function. If I want to plot one point per grain (due to grain size heterogeneity for instance), which option is better?..
```
odf_gl = calcDensity(ebsd('g').orientations,'halfwidth',10*degree);

PFs_gl_hkl = [ ...
    Miller(1,0,0,ebsd('g').CS,'hkl'), ...
    Miller(0,1,0,ebsd('g').CS,'hkl'), ...
    Miller(0,0,1,ebsd('g').CS,'hkl'), ...
    ]
%% Pole figure ODF
hold on
plotPDF(odf_gl,PFs_gl_hkl,'antipodal','MarkerSize',1)
colorbar
annotate([xvector,yvector,zvector],...
'label',{'X','Z','Y'},'backgroundcolor','w','MarkerSize',20,'FontSize',20)
mtexColorbar
drawNow(gcm,'figSize','large')
hold off

 %% Pole figure meanOrientation
figure (2)
plotPDF(grains('g').meanOrientation,PFs_gl_hkl,'antipodal','resolution',5*degree,'MarkerSize',1,'contourf');
hold on
plotPDF(grains('g').meanOrientation,PFs_gl_hkl,'MarkerSize',1); %
colorbar
annotate([xvector,yvector,zvector],...
'label',{'X','Z','Y'},'backgroundcolor','w','MarkerSize',20,'FontSize',20)
mtexColorbar
drawNow(gcm,'figSize','large')
hold off 
```

This one is for the ODF-derived PF
![image](https://user-images.githubusercontent.com/76631957/111645382-f105d180-8800-11eb-8050-76353f6cfb52.png)
And this one using meanOrientation
![image](https://user-images.githubusercontent.com/76631957/111645607-2ad6d800-8801-11eb-963f-7eb124952516.png)

Why are 010 orientations so different?

**Comment by kilir** [comment]
Hi,
a) you should probably derive both from an odf which you can calculate from either all orientations (what you did at the top) or from grains.meanOrientation. In both cases, guessing a kernel using calcKernel() might not be a bad idea.
```
psi = calcKernel(grains('g').meanOrientation)
odf_1 = calcDensity(ebsd('g').orientations,'kernel',psi);
odf_2 = calcDensity(grains('g').meanOrientation,'kernel',psi);
```
The difference in the (010) most likely stems from the fact that you have some large grains with a different orientation than most others.

Whatever is the best or correct one (number or volume weighted) depends on the question you want to answer.


Cheers,
Rüdiger

**Reply by Jesusmunozmontecinos** [reply]
Thanks a lot!!
